include/xtensor/xadapt.hpp

```diff
@@ -209,7 +209,8 @@ namespace xt
      * @param container the container to adapt
      * @param l the layout_type of the xtensor_adaptor
      */
-    template <layout_type L = XTENSOR_DEFAULT_LAYOUT, class C>
+    template <layout_type L = XTENSOR_DEFAULT_LAYOUT, class C,
+              XTL_REQUIRES(detail::not_an_array<std::decay_t<C>>)>
     inline xtensor_adaptor<C, 1, L>
```

Comment:
I think the return type should be `xtensor_adaptor<xtl::closure_type_t<C>, 1, L>`, according to the implementation below.

Reply:
yep, but I didn't really want to change existing implementation, as I'm not too deep into it)